js/src/api/contract/contract.js

```diff
@@ -29,7 +29,9 @@ export default class Contract {
     }
 
     this._api = api;
-    this._abi = new Abi(abi);
+    this._abi = abi instanceof Abi
```

Comment:
I prefer the original approach where we are explicit on what we expect in as opposed to having either one or the other? No other way around it?